e2e_samples/unstructured_data/docs/experiments-and-evaluations.md

```diff
@@ -0,0 +1,300 @@
+
+```text
+└── experiments
+  └── <experiment_name>
+    ├── main.py -> the entry point to the experiment
+    └──  config
+      ├── experiment.yaml
+      └── variants -> the variants directory, containing all variant config files
+        ├── <...variant-config-files>.yaml
+```
+
+> Note: For the `llm_citation_generator`, the variants directory is defined as `questions`.
+
+#### Experiment Classes
+
+An experiment is a Python class that has defined `__call__` and `__init__` methods.
+
+Requirements:
+
+- `__init__` method must accept `**kwargs` or `run_id` and `variant_name` because the orchestrator passes these values as keyword arguments to each experiment.
+- `__call__` method must accept `**kwargs` or the top-level values listed in each line of the data file that is passed to the orchestrator.
+
+Example:
+
+```python
+class Experiment:
+    def __init__(self, question: str, run_id: str, **kwargs):
+        self.question = question
+        self.run_id = run_id
+
+    def __call__(self, submission_folder: str, **kwargs):
+        ...
+```
+
+This experiment would expect a data file that looks like the following:
+
+```json
+{"submission_folder": "path/to/submission-1"}
+{"submission_folder": "path/to/submission-2"}
+{"submission_folder": "path/to/submission-3"}
+```
+
+The experiment would also expect a variant that defines `init_args.question` in the YAML definition. The class is initialized once for each variant.
+
+For each data line, the experiment's call method would be invoked using the data from that line. To support evaluations, the data file can also contain additional evaluation data in each line. For this reason, a `__call__` method should accept `**kwargs`, so they can be ignored if included in the data file. All input data from this file is included in the experiment results output for each data line in a flat list.
+
+Example data file with truth:
+
+```json
+{"submission_folder": "path/to/submission-1", "truth": {"total-revenue": "100", "earnings-per-share": "1.11"}}
+{"submission_folder": "path/to/submission-2", "truth": {"total-revenue": "150", "earnings-per-share": "2.22"}}
+{"submission_folder": "path/to/submission-3", "truth": {"total-revenue": "125", "earnings-per-share": "3.33"}}
+```
+
+Example output of the experiment results:
+
+```json
+{"inputs.submission_folder": "path/to/submission-1", "inputs.truth.total-revenue": "100", "inputs.truth.earnings-per-share": "1.11", "citations": ["citation1", "citation2"]}
+{"submission_folder": "path/to/submission-2", "truth": {"inputs.truth.total-revenue": "150", "inputs.truth.earnings-per-share": "2.22"}, "citations": ["citation1"]}
+{"submission_folder": "path/to/submission-3", "truth": {"inputs.truth.total-revenue": "125", "inputs.truth.earnings-per-share": "3.33"}, "citations": ["citation1", "citation2", "citation3"]}
+```
+
+#### Experiment Configuration
```

Comment:
If the evaluators config is optional here, can we specifically write `optional` in front of the description of the key-value pairs in this example